mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -274,15 +274,25 @@ public MessageMetaData add(Mailbox mailbox, MailboxMessage message) throws Mailb
     }
 
     private MailboxMessage addUidAndModseq(MailboxMessage message, CassandraId mailboxId) throws MailboxException {
-        final Mono<MessageUid> messageUidMono = uidProvider.nextUid(mailboxId).cache();
-        final Mono<ModSeq> nextModSeqMono = modSeqProvider.nextModSeq(mailboxId).cache();
-        Flux.merge(messageUidMono, nextModSeqMono).then();
-
-        message.setUid(messageUidMono.blockOptional()
-            .orElseThrow(() -> new MailboxException("Can not find a UID to save " + message.getMessageId() + " in " + mailboxId)));
-        message.setModSeq(nextModSeqMono.blockOptional()
-            .orElseThrow(() -> new MailboxException("Can not find a MODSEQ to save " + message.getMessageId() + " in " + mailboxId)));
-
+        Mono<MessageUid> messageUidMono = uidProvider
+            .nextUid(mailboxId)
+            .switchIfEmpty(Mono.error(() -> new MailboxException("Can not find a UID to save " + message.getMessageId() + " in " + mailboxId)));
+
+        Mono<ModSeq> nextModSeqMono = modSeqProvider.nextModSeq(mailboxId)
+            .switchIfEmpty(Mono.error(() -> new MailboxException("Can not find a MODSEQ to save " + message.getMessageId() + " in " + mailboxId)));
+
+        try {
+            Mono.zip(messageUidMono, nextModSeqMono)
+                .doOnNext(tuple -> {
+                    message.setUid(tuple.getT1());
+                    message.setModSeq(tuple.getT2());
+                })
+                .block();
```

Comment:
maybe it as not its place in this PR, but this method could return a Mono, and the `add` method flatmap over its result

Reply:
Yes, I tried to focus on the inspections results for this PR